src/fastga_he/powertrain_builder/powertrain.py

```diff
@@ -216,19 +108,24 @@ def load(self, power_train_file):
         if not FASTGAHEPowerTrainConfigurator._cache.get(self._power_train_file):
             FASTGAHEPowerTrainConfigurator._cache[self._power_train_file] = {}
 
-        self._serializer = _YAMLSerializer()
-        self._serializer.read(self._power_train_file)
```

Comment:
From what we discussed, this is the part that really needs to be cached, the rest is not that expensive from your last PR. So let's do it little by little